unstructured/ingest/connector/weaviate.py

```diff
@@ -0,0 +1,134 @@
+    BaseConnectorConfig,
+    BaseDestinationConnector,
+    BaseIngestDoc,
+    WriteConfig,
+)
+from unstructured.ingest.logger import logger
+from unstructured.utils import requires_dependencies
+
+
+@dataclass
+class SimpleWeaviateConfig(BaseConnectorConfig):
+    host_url: str
+    class_name: str
+    auth_keys: t.Optional[t.Dict[str, str]] = None
+
+
+@dataclass
+class WeaviateWriteConfig(WriteConfig):
+    batch_size: int = 100
+
+
+@dataclass
+class WeaviateDestinationConnector(BaseDestinationConnector):
+    write_config: WeaviateWriteConfig
+    connector_config: SimpleWeaviateConfig
+
+    @requires_dependencies(["weaviate"], extras="weaviate")
+    def initialize(self):
+        from weaviate import Client
+
+        auth = self._resolve_auth_method()
+
+        self.client: Client = Client(
+            url=self.connector_config.host_url,
+            auth_client_secret=auth,
+        )
+
+    def _resolve_auth_method(self):
+        if self.connector_config.auth_keys is None:
+            return None
+
+        if access_token := self.connector_config.auth_keys.get("access_token"):
+            from weaviate.auth import AuthBearerToken
+
+            return AuthBearerToken(
+                access_token=access_token,
+                refresh_token=self.connector_config.auth_keys.get("refresh_token"),
+            )
+        elif api_key := self.connector_config.auth_keys.get("api_key"):
+            from weaviate.auth import AuthApiKey
+
+            return AuthApiKey(api_key=api_key)
+        elif client_secret := self.connector_config.auth_keys.get("client_secret"):
+            from weaviate.auth import AuthClientCredentials
+
+            return AuthClientCredentials(
+                client_secret=client_secret, scope=self.connector_config.auth_keys.get("scope")
+            )
+        elif (username := self.connector_config.auth_keys.get("username")) and (
+            pwd := self.connector_config.auth_keys.get("password")
+        ):
+            from weaviate.auth import AuthClientPassword
+
+            return AuthClientPassword(
+                username=username, password=pwd, scope=self.connector_config.auth_keys.get("scope")
+            )
+        return None
+
+    def conform_dict(self, element: dict) -> None:
+        """
+        Updates the element dictionary to conform to the Weaviate schema
+        """
+
+        if (
+            record_locator := element.get("metadata", {})
+            .get("data_source", {})
+            .get("record_locator")
+        ):
+            # Explicit casting otherwise fails schema type checking
+            element["metadata"]["data_source"]["record_locator"] = str(json.dumps(record_locator))
+
+        if (
+            date_modified := element.get("metadata", {})
+            .get("data_source", {})
+            .get("date_modified", None)
+        ):
+            element["metadata"]["data_source"]["date_modified"] = date_modified + "Z"
```

Comment:
Wouldn't Weaviate support most of [these](https://github.com/Unstructured-IO/unstructured/blob/9fd77a523207fe7845ede28444ed361784e7783e/unstructured/ingest/connector/azure_cognitive_search.py#L48-L89) fields

As many of those fields that Weaviate supports, it might be good to include all of them here